app/models/pro_account.rb

```diff
@@ -38,6 +38,7 @@ def stripe_customer
   def update_stripe_customer
     return unless feature_enabled?(:pro_pricing)
 
+    @subscriptions = nil unless stripe_customer
```

Comment:
Not 100% sure what the problem is without this – could you post a quick code example of a before/after?

Reply:
This is the key to this solution.

After cancelling the incomplete subscriptions in the controller `@subscriptions` would be set for this ProAccount instance - it would be a SubscriptionCollection object which has been passed the current Stripe::Customer object. But at this point users might not have a Stripe::Customer object as this is before we add the card and the customer is save.

Without a customer in SubscriptionCollection#build attempts to create a Stripe::Subscription without a customer which obviously fails.

So setting to `nil` here allows SubscriptionCollection to be recreated with the new Stripe::Customer when ProAccount#subscriptions is next called.